e2e/tests/core/03-connection/connection_test.go

```diff
@@ -0,0 +1,89 @@
+package connection
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	paramsproposaltypes "github.com/cosmos/cosmos-sdk/x/params/types/proposal"
+	"github.com/strangelove-ventures/ibctest/ibc"
+	"github.com/strangelove-ventures/ibctest/test"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/cosmos/ibc-go/e2e/testsuite"
+	"github.com/cosmos/ibc-go/e2e/testvalues"
+	transfertypes "github.com/cosmos/ibc-go/v5/modules/apps/transfer/types"
+	connectiontypes "github.com/cosmos/ibc-go/v5/modules/core/03-connection/types"
+	ibctesting "github.com/cosmos/ibc-go/v5/testing"
+)
+
+func TestConnectionTestSuite(t *testing.T) {
+	suite.Run(t, new(ConnectionTestSuite))
+}
+
+type ConnectionTestSuite struct {
+	testsuite.E2ETestSuite
+}
+
+// QueryConnectionEnabledParam queries the on-chain connection enabled param for 03-connection
+func (s *ConnectionTestSuite) QueryMaxExpectedTimePerBlockParam(ctx context.Context, chain ibc.Chain) uint64 {
+	queryClient := s.GetChainGRCPClients(chain).ParamsQueryClient
+	res, err := queryClient.Params(ctx, &paramsproposaltypes.QueryParamsRequest{
+		Subspace: "ibc",
+		Key:      string(connectiontypes.KeyMaxExpectedTimePerBlock),
+	})
+	s.Require().NoError(err)
+
+	// removing additional strings that are used for amino
+	delay := strings.ReplaceAll(res.Param.Value, "\"", "")
+	time, err := strconv.ParseUint(delay, 10, 64)
+	s.Require().NoError(err)
+
+	return time
+}
+
+// TestMaxExpectedTimePerBlock tests changing the MaxExpectedTimePerBlock param using a governance proposal
+func (s *ConnectionTestSuite) TestMaxExpectedTimePerBlock() {
+	t := s.T()
+	ctx := context.TODO()
+
+	_, _ = s.SetupChainsRelayerAndChannel(ctx, transferChannelOptions())
+	chainA, chainB := s.GetChains()
+	chainAWallet := s.CreateUserOnChainA(ctx, testvalues.StartingTokenAmount)
+
+	s.Require().NoError(test.WaitForBlocks(ctx, 1, chainA, chainB), "failed to wait for blocks")
+
+	t.Run("ensure delay is set to the default of 30 seconds", func(t *testing.T) {
+		expectedDelay := fmt.Sprintf("\"%d\"", 30*time.Second)
```

Comment:
rather than escaping double quotes `\"` characters, I like to use template strings to make it more readable.

backticks don't render nicely in markdown (or at least I can't figure out how to make them render)

see [this example](https://go.dev/play/p/KqYnXetLlj8)